src/clj/opentracing_clj/core.clj

```diff
@@ -180,35 +215,21 @@
   (let [s (bindings 0)
         m (bindings 1)]
     `(let [m#  ~m
-           st# (s/conform :opentracing/span-init m#)]
-       (cond (= :clojure.spec.alpha/invalid st#)
-             (throw (ex-info "with-span binding failed to conform to :opentracing/span-init"
-                             (s/explain-data :opentracing/span-init m#)))
-
-             (= :new (first st#))
-             (let [sb# (sb/build-span *tracer* (:name m#))]
-               (when-let [tags# (:tags m#)]
-                 (sb/add-tags sb# tags#))
-               (when (:ignore-active? m#)
-                 (sb/ignore-active sb#))
-               (when-let [start-ts# (:start-timestamp m#)]
-                 (sb/with-start-timestamp sb# start-ts#))
-               (when-let [parent# (:child-of m#)]
-                 (sb/child-of sb# parent#))
-               (with-open [^Scope scope# (sb/start sb# (or (nil? (:finish? m#))
-                                                           (:finish? m#)))]
-                 (let [~s (.span scope#)]
-                   ~@body)))
-
-             (= :existing (first st#))
-             (with-open [^Scope scope# (.activate (.scopeManager *tracer*) (:from m#)
-                                                  (or (nil? (:finish? m#))
-                                                      (:finish? m#)))]
-               (let [~s (.span scope#)]
-                 ~@body))
-
-             :else
-             (throw (ex-info "Unknown error." {}))))))
+           ~s  (get-span m#)]
+       (with-open [^Scope _# (.activate (.scopeManager *tracer*)
+                                        ~s)]
+         (try
+           ~@body
+           (catch Exception e#
```

Comment:
This is the part that I'm most iffy on based on personal opinion.  I want to try to avoid locking people to a certain implementation if it can be helped even if it followed opentracing's semantic conventions.  Would be more open to having this, if it was a default convention that can be overridden.

I think it would be best however to keep the exception handling the responsibility of the developer.  It's a little messier on their end to have opentracing statements littering their catch clauses, but exceptions will have to be handled anyway inside the body.

Reply:
I have added the option `:process-exceptions?` as part of the span-init.  Please let me know what you think.

Reply:
I think that's better but still have some reservations and could use your input.  

First, I think if we're keeping this, maybe we should go ahead and add the stack log field as well?  IIRC, `error.object` doesn't include the stacktrace when it's converted to a string for viewing. 

Second, there's the question of whether we should take the stance of treating exceptions as errors by default.  Personally, I'm not sure of this but can be convinced otherwise.  An argument can be made that if exceptions are seen foremost as a means of control flow, then an uncaught exception associated with an operation might just signal the end of that operation.  So if we do take the argument that exceptions are not necessarily signalling errors, we'd have to reconsider whether we should make this the default behaviour (or whether we keep the behaviour at all).

Reply:
I think for now we can drop the exception feature and you can make a separate pull request for it so that we can at least merge the changes regarding the dependency updates.

Reply:
Haven't seen activity here in a while so will be merging this PR soon and then making the changes on dropping the exception feature.  Again, feel free to open another pull request to continue that conversation.